includes/backends/class-omise-backend-installment.php

```diff
@@ -0,0 +1,124 @@
+				'min_allowed_amount' => 300.00,
+			),
+
+			'installment_bay' => array(
+				'bank_code'          => 'bay',
+				'title'              => __( 'Krungsri', 'omise' ),
+				'interest_rate'      => 0.008,
+				'min_allowed_amount' => 300.00,
+			),
+
+			'installment_ktc' => array(
+				'bank_code'          => 'ktc',
+				'title'              => __( 'Krungthai Card (KTC)', 'omise' ),
+				'interest_rate'      => 0.008,
+				'min_allowed_amount' => 300.00,
+			),
+
+			'installment_bbl' => array(
+				'bank_code'          => 'bbl',
+				'title'              => __( 'Bangkok Bank', 'omise' ),
+				'interest_rate'      => 0.008,
+				'min_allowed_amount' => 500.00,
+			),
+
+			'installment_kbank' => array(
+				'bank_code'          => 'kbank',
+				'title'              => __( 'Kasikorn Bank', 'omise' ),
+				'interest_rate'      => 0.0065,
+				'min_allowed_amount' => 500.00,
+			),
+		);
+	}
+
+	/**
+	 * @param  string $currency
+	 * @param  float  $purchase_amount
+	 *
+	 * @return array  of an available installment providers
+	 */
+	public function get_available_providers( $currency, $purchase_amount ) {
+		// Note: as installment payment at the moment only supports for THB currency, so the 
+		//       $purchase_amount is multiplied with 100 to convert the amount into subunit (satang).
+		$providers = $this->capabilities()->getInstallmentBackends( $currency, ( $purchase_amount * 100 ) );
+
+		foreach ( $providers as &$provider ) {
+			$provider->provider_code             = str_replace( 'installment_', '', $provider->_id );
+			$provider->provider_name             = isset( self::$providers[ $provider->_id ] ) ? self::$providers[ $provider->_id ]['title'] : strtoupper( $provider->code );
+			$provider->allowed_installment_terms = $this->filter_available_terms( $provider->allowed_installment_terms, self::$providers[ $provider->_id ], $purchase_amount );
+			$provider->interest                  = $this->capabilities()->is_zero_interest() ? 0 : ( self::$providers[ $provider->_id ]['interest_rate'] * 100 );
+		}
+
+		return $providers;
+	}
+
+	/**
+	 * @param  array $available_terms
+	 * @param  array $provider_detail
+	 * @param  float $purchase_amount
+	 *
+	 * @return array  of an filtered available terms
+	 */
+	public function filter_available_terms( $available_terms, $provider_detail, $purchase_amount ) {
+		$filtered_available_terms = array();
+
+		sort( $available_terms );
+
+		for ( $i = 0; $i < count( $available_terms ); $i++ ) {
+			$term           = $available_terms[ $i ];
+			$monthly_amount = $this->calculate_monthly_payment_amount(
+				$purchase_amount,
+				$term,
+				$this->capabilities()->is_zero_interest() ? 0 : $provider_detail['interest_rate']
+			);
+
+			if ( $monthly_amount < $provider_detail['min_allowed_amount'] ) {
+				break;
+			}
+
+			$filtered_available_terms[ $i ] = array(
+				'term'           => $term,
+				'monthly_amount' => $monthly_amount
+			);
+		}
+
+		return $filtered_available_terms;
```

Comment:
After reading the whole function it seems really that it is possibly poorly named - as it is transforming the array to have different content, as well as sorting and filtering.

Reply:
@jonrandy do you have any suggestion name? Maybe I use `filter` in a wrong meaning here. 
The intention of the code is that you throw the whole list of all possible terms in, then get only available terms that matched with the requirement out.

Reply:
Well, the `type` of what is in the array coming out here is different to what is going in - so it is doing more than just filtering. It's also sorting. Maybe something like 'get_valid_term_options` would be better, where we consider a `term_option` to be the array containing the number of `terms` (plural important) and the `monthly_amount`.

Alternatively if you want to go all functional, you could have one function to transform a `term` to a `term_option`, and another to check if a term option was valid... then, in pseudocode:

```
availableTerms = terms.sort().map(convertToTermOptions).filter(validTermOption)
```

Although this isn't as efficient as your code above correctly stops adding any more valid term options as soon as it hits the first one that is below minimum

Reply:
@jonrandy Thanks for the explanation 👍 I see your point why `filter` doesn't make sense now.
Just wonder would it be any confusion between `allowed`, `available`, and `valid`?

Reply:
I have no problem with any of those terms, it wasn't a conscious decision to change to to `valid`